### PR TITLE
Remove dependency on unittest2

### PR DESCRIPTION
**Pull request by mcepl**
It is questionable whether it is actually the right way how to do it.
I would need a review from somebody who actually understands API and the
protocol, whether I am not unintentionally changing API.

### CODE REVIEW

**Comment by mcepl** [comment]
It is possible, that there is actually some problem with this change, because I get testing-cabal/testtools#121 so I am probably missing something.

File: python/subunit/tests/test_test_protocol.py [hunk 1]
```diff
@@ -16,10 +16,10 @@
 
 import datetime
 import io
-import unittest2 as unittest
 import os
 import sys
 import tempfile
+import unittest
```
[the remark this answers is not in the document]
Comment (mcepl) [reply python/subunit/tests/test_test_protocol.py]: Why not
```python
from testtools.testcase import unittest
```
???

File: python/subunit/tests/test_test_protocol.py [hunk 2]
```diff
@@ -152,13 +152,20 @@ def test_story(self):
...
         bing = subunit.RemotedTestCase("bing crosby")
         an_error = subunit.RemotedTestCase("an error")
-        self.assertEqual(
-            client.errors,
-            [(an_error, tb_prelude + _remote_exception_repr + '\n')])
-        self.assertEqual(
-            client.failures,
-            [(bing, tb_prelude + _remote_exception_repr + ": "
-              + details_to_str({'traceback': text_content(traceback)}) + "\n")])
+        if sys.version_info[0] >= 3:
```
[the remark this answers is not in the document]
Comment (mcepl) [reply python/subunit/tests/test_test_protocol.py]: Hmm, ``testtools`` requires ``six``, so it is already indirect dependency of subunit anwyay. Shouldn’t we just use it?

**Comment by graingert** [comment]
@jelmer @mtreinish this is blocking subunit support in Twisted on Python 3.10

My current approach is to disable subunit on twisted on py3.10 but I'd very much like to re-enable it before 3.10.0 is released

**Comment by jelmer** [comment]
I don't see a good reason to block this, so let's just land it.

**Pull request merged**